docs/tofu.logging.recipes.zio.md

```diff
@@ -0,0 +1,168 @@
+{"@timestamp":"2021-09-19T21:12:18.017Z","loggerName":"BarServiceImpl","threadName":"zio-default-async-4","level":"DEBUG","message":"Got current date 2021-09-20"}
+```
+### Custom `Loggable`
+Default `Loggable[LocalDate]` is based on `stringValue` instance, hence the date was logged as a plain string, not key-value. 
+What if you want the date to be a separated field in json? Well, you can customize `Loggable`. Add the following line into the `BarServiceImpl`:
+```scala
+private implicit val customDateLoggable = Loggable[LocalDate].named(name="date")
+```
+Now the output looks like:
+```json lines
+{"@timestamp":"2021-09-19T21:12:18.017Z","loggerName":"BarServiceImpl","threadName":"zio-default-async-4","level":"DEBUG","message":"Got current date 2021-09-20","date":"2021-09-20"}
+```
+`Loggable.apply[LocalDate]` summons an instance from the global scope, `.named` converts logged object into a single field named `name`.
+There are several methods to create and modify `Loggable` instances, read more in [Loggable](./tofu.logging.loggable.md) section.
+
+### Context logging
+Let's consider how to log the context along with actual log message. If you have an instance of `Loggable` for your context, 
+you can have it logged automagically. For example:
+```scala
+import derevo.derive
+import tofu.logging.derivation.loggable
+
+@derive(loggable)
+final case class Context(requestId: Int)
+```
+Here we use [Tofu Derevo](https://github.com/tofu-tf/derevo) for automatic derivation `Loggable` instance.
+
+The main idea of this approach is to store your in ZIO [FiberRef](https://zio.dev/docs/datatypes/fiber/fiberref). 
+It provides all the power of State Monad. Unlike Java's `ThreadLocal`, `FiberRef` has copy-on-fork semantic: 
+a child [Fiber](https://zio.dev/docs/datatypes/fiber/fiber/) starts with `FiberRef` values of its parent.
+When the child set a new value of FiberRef, the change is visible only to the child itself. This means if we set `requestId` value to `117`
+(e.g. at the start of the request) and pass the `FiberRef` to a child fiber, it sees the value `117`.
+
+Let's modify Main app.  We have to define context service and use `layerPlainWithContext`. 
+```scala
+object Main extends zio.App {
+  val contextLayer: ULayer[Has[FiberRef[Context]]] = FiberRef.make(Context(-10)).toLayer
+
+  val logsLayer: URLayer[Has[FiberRef[Context]], TofuLogs] = ZLogging.Make.layerPlainWithContext(_.get)
+
+  override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] = {
+    for {
+      (barService, ref) <- ZIO.services[BarService, FiberRef[Context]]
+      f1 <- barService.foo.fork
+      _ <- ref.set(Context(117))
+      f2 <- barService.foo.fork
+      _ <- f1.join <&> f2.join
+    } yield ()
+  }.injectCustom(
+    BarService.live,
+    logsLayer,
+    contextLayer
+  ).exitCode
+}
+```
+Run the program and look at the output:
+```json lines
```

Comment:
Perhaps it's better to write this example using ConsoleContextLayout for readability. Even on 23'' screen I have to scroll it right now to see the main stuff.

Reply:
I think it's ok just remove extra fields